### PR TITLE
feat(combobox): add nested multiselect example

### PR DESCRIPTION
**Pull request by jzempel**
## Description

The new controlled example considers the following:
- in place `<Option>` swapping based on key next/previous values
- ~usage of root `<Option>` icons to indicate sub-option selection~ (TBD in v9)
- always rendering all selected options, including `<Option isHidden>` for accessibility
- a listbox collapse effect to restore root `<Option>` state
- controlling `activeIndex` for improved "skip past sub-header option" ergonomics

Screenshot 2024-09-06 at 11 05 41 AM

## Detail

Also adds similar enhancements to the existing nested (single select) example.

## Checklist

- [x] :ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)
- [x] :black_nib: copy updates are approved (add the content strategist as a reviewer)
- [x] :link: considered opportunities for adding cross-reference URLs (grep for keywords)
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :memo: tested in Chrome, Firefox, Safari, Edge

### CODE REVIEW

File: src/examples/combobox/NestedMultiselect.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+interface IOption extends IOptionProps {
+  value: string;
+}
+interface IOptGroup extends IOptGroupProps {
+  options: IOption[];
+}
+
+type Options = (IOption | IOptGroup)[];
+
+type SubOptions = Record<any, Options>;
```
Comment (ze-flo) [inline src/examples/combobox/NestedMultiselect.tsx:32]: [nit]

```suggestion
type SubOptions = Record<string, Options>;
```

File: src/examples/combobox/NestedMultiselect.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        const subValues = option.options.map(subOption => subOption.value);
+
+        _values.push(...subValues);
+      }
+
+      return _values;
+    }, []);
+    const missingValues = state.selectionValue.filter(value => !values.includes(value));
+
+    return missingValues.map(value => <Option key={value as string} isHidden value={value} />);
```
Comment (ze-flo) [inline src/examples/combobox/NestedMultiselect.tsx:143]: Reminder: the casting `key={value as string}` is redundant with v9.

**Review by ze-flo — APPROVED** [review]
Beautiful work! ✨

File: src/examples/combobox/NestedMultiselect.tsx (hunk 1 above)
Comment (jzempel) [reply src/examples/combobox/NestedMultiselect.tsx:32]: @ze-flo we'll get these type enhancements dialed-in for v9, where `OptionValue` is only valid for `string`. In order to let the `OptionValue = string | object` flow through the example, this has to be `any` 😞

File: src/examples/combobox/NestedMultiselect.tsx (hunk 2 above)
Comment (jzempel) [reply src/examples/combobox/NestedMultiselect.tsx:143]: Yep, the v9 example is going to look cleaner ✨

**Review by geotrev — APPROVED** [review]
Non-blocking question because this would (presumably) be a concern on the container side: should an `OptGroup` with nested options respond to arrow right (or left, in RTL) to dig into nested options (similar to nested menu items)?

**Comment by ze-flo** [comment]
To avoid merge conflicts, I'll handle the Renovate + Dependabot PRs once this one merges.

**Comment by jzempel** [comment]
> To avoid merge conflicts, I'll handle the Renovate + Dependabot PRs once this one merges.

@ze-flo we're deferring all of the bot updates until after `next` -> `main`.

**Comment by jzempel** [comment]
> Non-blocking question because this would (presumably) be a concern on the container side: should an `OptGroup` with nested options respond to arrow right (or left, in RTL) to dig into nested options (similar to nested menu items)?

Yeah, we should explore the container enhancement at some point. I think the question is how things will work with/against `downshift` :shrug:

**Review by steue — APPROVED** [review]
LGTM so far. looking forward to our improvement for the top level multi selection indicator

**Comment by jzempel** [comment]
> Just noticed there is a small input bug when trying to backspace, the first character is never deleted.
> 
>  Screen.Recording.2024-09-11.at.9.28.40.AM.mov

Thanks, @geotrev. Those or conditions never quite work out like they look on paper 😜. Changed `changes.inputValue || state.inpuValue` to `changes.inputValue === undefined ? state.inputValue : changes.inputValue`

**Pull request merged**